packages/deployments/contracts/contracts/facets/StableSwapFacet.sol

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+import {BaseConnextFacet} from "./BaseConnextFacet.sol";
+import {AmplificationUtils, SwapUtils} from "../libraries/AmplificationUtils.sol";
+import {LPToken} from "../libraries/LPToken.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+
+/**
+ * @title Swap - A StableSwap implementation in solidity.
+ * @notice This contract is responsible for custody of closely pegged assets (eg. group of stablecoins)
+ * and automatic market making system. Users become an LP (Liquidity Provider) by depositing their tokens
+ * in desired ratios for an exchange of the pool token that represents their share of the pool.
+ * Users can burn pool tokens and withdraw their share of token(s).
+ *
+ * Each time a swap between the pooled tokens happens, a set fee incurs which effectively gets
+ * distributed to the LPs.
+ *
+ * In case of emergencies, admin can pause additional deposits, swaps, or single-asset withdraws - which
+ * stops the ratio of the tokens in the pool from changing.
+ * Users can always withdraw their tokens via multi-asset withdraws.
+ *
+ * @dev Most of the logic is stored as a library `SwapUtils` for the sake of reducing contract's
+ * deployment size.
+ */
+
+contract StableSwapFacet is BaseConnextFacet {
+  using SwapUtils for SwapUtils.Swap;
+  using AmplificationUtils for SwapUtils.Swap;
+
+  // ========== Custom Errors ===========
+  error StableSwapFacet__deadlineCheck_deadlineNotMet();
+  error StableSwapFacet__initializeSwap_alreadyInitialized();
+  error StableSwapFacet__initializeSwap_invalidPooledTokens();
+  error StableSwapFacet__initializeSwap_decimalsMismatch();
+  error StableSwapFacet__initializeSwap_duplicateTokens();
+  error StableSwapFacet__initializeSwap_zeroTokenAddress();
+  error StableSwapFacet__initializeSwap_tokenDecimalsExceedMax();
+  error StableSwapFacet__initializeSwap_aExceedMax();
+  error StableSwapFacet__initializeSwap_feeExceedMax();
+  error StableSwapFacet__initializeSwap_adminFeeExceedMax();
+  error StableSwapFacet__initializeSwap_failedInitLpTokenClone();
+  error StableSwapFacet__getToken_outOfRange();
+  error StableSwapFacet__getTokenIndex_notExist();
+  error StableSwapFacet__getTokenBalance_indexOutOfRange();
+
+  // ============ Properties ============
+
+  // ============ Modifiers ============
+
+  /**
+   * @notice Modifier to check deadline against current timestamp
+   * @param deadline latest timestamp to accept this transaction
+   */
+  modifier deadlineCheck(uint256 deadline) {
+    if (block.timestamp > deadline) revert StableSwapFacet__deadlineCheck_deadlineNotMet();
+    _;
+  }
+
+  /*** VIEW FUNCTIONS ***/
```

Comment:
Lol ik this is such a nit but:
```
// ============ Modifiers ============
```
vs
```
 /*** VIEW FUNCTIONS ***/
 ```
 
 can we make these the same vibe

Reply:
e.g.
```
// ========= View Functions =========
```

Reply:
actually nvm, let's just come thru later and polish comments and stuff

Reply:
Okay ~